src/framework/theme/theme.spec.tsx

```diff
@@ -4,45 +4,95 @@ import { render } from 'react-native-testing-library';
   }
 }
 
-it('Checks theme consumer renders properly', async () => {
-  const ThemedComponent = withTheme(TestComponent);
-  const themedComponentTestId = '@theme/root';
+describe('@theme: theme consumer checks', () => {
+
+  it('renders properly', async () => {
+    const ThemedComponent = withTheme(TestComponent);
+
+    const component = render(
+      <ThemeProvider theme={{}}>
+        <ThemedComponent testId={themeConsumerTestId}/>
+      </ThemeProvider>,
+    );
+
+    const themedComponent = component.getByTestId(themeConsumerTestId);
+    expect(themedComponent).not.toBeNull();
+  });
 
-  const component = render(
-    <ThemeProvider theme={{}}>
-      <ThemedComponent testId={themedComponentTestId}/>
-    </ThemeProvider>,
-  );
+  it('receives theme prop', async () => {
+    const ThemedComponent = withTheme(TestComponent);
+
+    const component = render(
+      <ThemeProvider theme={{}}>
+        <ThemedComponent testId={themeConsumerTestId}/>
+      </ThemeProvider>,
+    );
+
+    const themedComponent = component.getByTestId(themeConsumerTestId);
+    expect(themedComponent.props.theme).not.toBeNull();
+  });
 
-  const themedComponent = component.getByTestId(themedComponentTestId);
-  expect(themedComponent).not.toBeNull();
 });
 
-it('Checks theme consumer receives theme prop', async () => {
-  const ThemedComponent = withTheme(TestComponent);
-  const themedComponentTestId = '@theme/root';
+describe('@theme: styled theme consumer checks', () => {
+
+  it('renders properly', async () => {
+    const ThemedComponent = withThemedStyles(TestComponent, (theme: ThemeShape) => {
+      return {};
+    });
+
+    const component = render(
+      <ThemeProvider theme={{}}>
+        <ThemedComponent testId={themeConsumerTestId}/>
+      </ThemeProvider>,
+    );
+
+    const themedComponent = component.getByTestId(themeConsumerTestId);
+    expect(themedComponent).not.toBeNull();
+  });
+
+  it('receives theme prop', async () => {
+    const ThemedComponent = withThemedStyles(TestComponent, (theme: ThemeShape) => {
+      return {};
+    });
+
+    const component = render(
+      <ThemeProvider theme={{}}>
+        <ThemedComponent testId={themeConsumerTestId}/>
+      </ThemeProvider>,
+    );
+
+    const themedComponent = component.getByTestId(themeConsumerTestId);
+    expect(themedComponent.props.theme).not.toBeNull();
+  });
+
+  it('receives themedStyle prop', async () => {
+    const ThemedComponent = withThemedStyles(TestComponent, (theme: ThemeShape) => {
+      return {};
+    });
+
+    const component = render(
+      <ThemeProvider theme={{}}>
+        <ThemedComponent testId={themeConsumerTestId}/>
+      </ThemeProvider>,
+    );
 
-  const component = render(
-    <ThemeProvider theme={{}}>
-      <ThemedComponent testId={themedComponentTestId}/>
-    </ThemeProvider>,
-  );
+    const themedComponent = component.getByTestId(themeConsumerTestId);
+    expect(themedComponent.props.themedStyle).not.toBeNull();
+  });
```

Comment:
Make sense to add tests which will check if the component will receive new theme when we change it.